noxfile.py

```diff
@@ -8,7 +8,7 @@
 DIR = Path(__file__).parent.resolve()
 
 nox.needs_version = ">=2024.3.2"
-nox.options.sessions = ["typecheck", "test_unit"]
+nox.options.sessions = ["typecheck", "tests"]
```

Comment:
Maybe we need to add a comment in the docs that for now `nox tests` will only run unit tests?

Reply:
This actually better matches my expectations -- I generally would expect pytest/nox to only run unit tests unless explicitly told to run integration tests as they are larger and require integrations. It's pretty common for unit tests to just be called "tests".

Reply:
@betolink, we can clarify the docs when we add integration tests to the mix with `nox` as per the recent issue you opened.

@jhkennedy, yep, agreed.

Perhaps the yet-to-be-added nox session could be named `integration_tests` (or `integration-tests`, if `nox` supports kebab-case names -- not sure if it does).